java/arcs/core/util/Scheduler.kt

```diff
@@ -259,6 +266,23 @@ class Scheduler(
         override fun iterator(): Iterator<Task> = listeners.values.flatten().iterator()
     }
 
+    /**
+     * Implementation of a [CoroutineDispatcher] which can be used by code unaware of the [Scheduler]
+     * itself to run operations as listeners on the scheduler.
+     *
+     * For example: [Handle.dispatcher] returns an instance of [Dispatcher] so that code outside of
+     * particle-context can run operations on the scheduler to access the handle data safely and
+     * correctly.
+     */
+    private class Dispatcher(private val scheduler: Scheduler) : CoroutineDispatcher() {
+        override fun dispatch(context: CoroutineContext, block: Runnable) =
+            scheduler.schedule(DispatchedTask(block::run))
+
+        private class DispatchedTask(
```

Comment:
What does "non-particle" mean in this context?

Reply:
users of the scheduler's dispatcher can always add a `CoroutineName` to the dispatcher context when using `withContext`.

E.g.

```kotlin
withContext(scheduler.asCoroutineDispatcher() + CoroutineName("my-thing")) {
    // do my thing
}
```